src/composables/useWarningScamMessage.ts

```diff
@@ -0,0 +1,33 @@
+import { createGlobalState } from '@vueuse/core';
+
+const useScamMessageStateStorage = createGlobalState(() => {
```

Comment:
We should use clear and appropriate naming (it's flagged not scam):
1. `useScamMessageStateStorage` -> `useMessageHiddenState`: Shifts focus from "scam" specifics to general message hiding.
2. `setScamMessages` -> `setMessageHidden`: Clarifies that we're setting hidden status, not just handling "scam" messages.
3. `useWarningScamMessage` -> `useFlaggedMessageHiddenState`: Broadens use from "scam" messages to any "flagged" messages.

Reply:
Maybe let's also use 'visible' instead of 'hidden'. This way, we focus on message visibility, not on hidden content. When message is visible content is hidden. Instead of when when content is hidden, message is visible. This is also more in line with the template structure: v-if message, v-else content".